src/custom-elements/src/components/date/date.tsx

```diff
@@ -76,11 +76,17 @@ export class Date {
   @Watch('value')
   dateChanged(newValue: string, oldValue: string) {
     if (newValue !== oldValue) {
```

Comment:
I don't know if stencil might have updated this, but in the previous versions, newValue and oldValue might have different values because newValue was null and oldValue was undefined. I think you should update your code to check this. 
```suggestion
    if (newValue !== oldValue && (newValue || oldValue)) {
```

Reply:
Please, don't solve this yet. It needs more checks.